DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/AnonymizerEngine.cs

```diff
@@ -0,0 +1,74 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using Dicom;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Anonymizer.Core.AnonymizerConfigurations;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Rules;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core
+{
+    public class AnonymizerEngine
+    {
+        private readonly ILogger _logger = AnonymizerLogging.CreateLogger<AnonymizerEngine>();
+        private readonly AnonymizerSettings _anonymizerSettings;
+        private readonly AnonymizerConfigurationManager _configurationManager;
+        private readonly IAnonymizerRuleFactory _ruleFactory;
+
+        public AnonymizerEngine(string configFilePath = "configuration.json", AnonymizerSettings anonymizerSettings = null, IAnonymizerRuleFactory ruleFactory = null)
+            : this(AnonymizerConfigurationManager.CreateFromConfigurationFile(configFilePath), anonymizerSettings, ruleFactory)
+        {
+        }
+
+        public AnonymizerEngine(AnonymizerConfigurationManager configurationManager, AnonymizerSettings anonymizerSettings = null, IAnonymizerRuleFactory ruleFactory = null)
```

Comment:
Consider not to parse config here, instead we can define method like:

Anonymizer.AnonymizeDataset(dataset, settings)

Reply:
Setting will be changed in AnonymizeDataset() function?